README.md

```diff
@@ -204,142 +206,30 @@ The rollup will call a bridge contract with a fixed deterministic amount of gas
-  function finalise(
-    AztecTypes.AztecAsset calldata _inputAssetA,
-    AztecTypes.AztecAsset calldata _inputAssetB,
-    AztecTypes.AztecAsset calldata _outputAssetA,
-    AztecTypes.AztecAsset calldata _outputAssetB,
-    uint256 _interactionNonce,
-    uint64 _auxData
-  )
-    external
-    payable
-    returns (
-      uint256 outputValueA,
-      uint256 outputValueB,
-      bool interactionComplete
-    );
-}
-
-```
-
-### Async flow explainer
-
-If a DeFi Bridge interaction is asynchronous, it must be finalised at a later date once the DeFi interaction has completed.
-
-This is achieved by calling `RollupProcessor.processAsyncDefiInteraction(uint256 interactionNonce)`.
-This internally will call `finalise` and ensure the correct amount of tokens has been transferred.
-
-#### convert(...)
-
-This function is called from the Aztec Rollup Contract via the DeFi Bridge Proxy.
-Before this function on your bridge contract is called the rollup contract will have sent you ETH or tokens defined by the input params.
-
-This function should interact with the DeFi protocol (e.g Uniswap) and transfer tokens or ETH back to the Aztec Rollup Contract.
-The Rollup contract will check it received the correct amount.
-
-If the DeFi Bridge interaction is asynchronous (e.g. it does not settle in the same block), the call to convert should return `(0,0 true)`.
-The contract should record the interaction nonce for any asynchronous position or if virtual assets are returned.
-
-At a later date, this interaction can be finalised by prodding the rollup contract to call `finalise` on the bridge.
-
-#### finalise(...)
-
-This function will be called from the Aztec Rollup contract.
-The Aztec Rollup contract will check that it received the correct amount of ETH and tokens specified by the return values and trigger the settlement step on Aztec.
+### Example flows and asset configurations
+
+1. Swapping - 1 real input, 1 real output
+2. Swapping with incentives - 1 real input, 2 real outputs (2nd output reward token)
+3. Borrowing - 1 real input (collateral), 1 real output (borrowed asset, e.g. Dai), 1 virtual output (represents the position, e.g. position is a vault in MakerDao)
+4. Purchasing an NFT - 1 real input, 1 virtual output (asset representing NFT)
+5. Selling an NFT - 1 virtual input (asset representing NFT), 1 real output (e.g. ETH)
+6. Repaying a loan - 1 real input (e.g. Dai), 1 virtual input (representing the vault/position), 1 real output (collateral, e.g. ETH)
+7. Repaying a loan 2 - 1 real input (e.g. USDC), 1 virtual input (representing the position), 2 real outputs ( 1st output collateral, 2nd output reward token, e.g. AAVE)
+8. Partial loan repaying - 1 real input (e.g. Dai), 1 virtual input (representing the vault/position), 1 real output (collateral, e.g. ETH), 1 virtual output (representing the vault/position)
+9. Claiming fees from Uniswap position - 1 virtual input (represents LP position NFT), 2 real outputs (fees in token0 and token1)
```

Comment:
You can exit the uniswap position with the virtual input. But to claim fees you would only be able to claim for one of them as you would need the virtual asset back.

Reply:
Addressed in [ca2056f](https://github.com/AztecProtocol/aztec-connect-bridges/pull/188/commits/ca2056f3ff8b39cf7c4c649c70e74b4285622ac0)